examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -459,37 +466,46 @@ def sim_update(self, dt: float, post_sim_update_dict):
 
         self._sps_tracker.increment()
 
+        for user_index in self._users.indices(Mask.ALL):
+            self._user_data[user_index].update(dt)
+            self._update_grasping_and_set_act_hints(user_index)
+            self._update_help_text(user_index)
+
         # Draw the picture-in-picture showing other agent's perspective.
         if self._users.max_user_count == 2:
             self._user_data[0].draw_pip_viewport(self._user_data[1])
             self._user_data[1].draw_pip_viewport(self._user_data[0])
 
-        if not self._paused:
-            for user_index in self._users.indices(Mask.ALL):
-                self._user_data[user_index].update(dt)
-                self._update_grasping_and_set_act_hints(user_index)
-            self._app_service.compute_action_and_step_env()
-
-        for user_index in self._users.indices(Mask.ALL):
-            self._update_help_text(user_index)
+        self._app_service.compute_action_and_step_env()
 
         # Set the server camera.
         server_cam_transform = self._user_data[
             self._server_user_index
         ].cam_transform
         post_sim_update_dict["cam_transform"] = server_cam_transform
 
+        #  Collect data.
+        self._elapsed_time += dt
+        if self._is_any_user_active():
+            # TODO: Add data collection.
+            pass
 
-@hydra.main(
-    version_base=None, config_path="config", config_name="rearrange_v2"
-)
-def main(config):
-    hitl_main(
-        config,
-        lambda app_service: AppStateRearrangeV2(app_service),
-    )
+    def _is_any_user_active(self) -> bool:
+        return any(
+            self._user_data[user_index].gui_input.get_any_input()
+            for user_index in range(self._app_data.max_user_count)
+        )
```

Comment:
How does this work for LLM-agent? Or do you not care? We probably should.

Reply:
This should work without changes with LLM agents. `user_data`, `max_user_count` and `user_index` all relate to human users.

In the LLM case, `max_user_count` will be `1`, but `agent_count` will be `2`. I have to decouple these concepts within `user_data` (will most likely create an analogous `agent_data` struct.